### PR TITLE
Added a version of t9 example with streaming terminals

### PR DESCRIPTION
**Pull request by pnookala**

### CODE REVIEW

File: examples/t9/t9_streaming.cc [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+    auto comp1 = make_compress(a, compa, "compress(A)");
+    auto &reduce_leaves_op = std::get<1>(comp1);
+
+    reduce_leaves_op->set_input_reducer<0>([](Node &&left, Node &&right) {
+                                           double s = (left.s + right.s) * 0.5;
+                                           double d = (left.s - right.s) * 0.5;
+                                           return Node(left.key, s, d, true);
+                                         });
+    //Key really doesn't matter here, but for compilation, we need to use the overload. FIX THIS!!
```
Comment (evaleev) [inline examples/t9/t9_streaming.cc:378]: @pnookala not following the issue ... is the comment still relevant?

File: ttg/ttg/madness/ttg.h [hunk 2]
```diff
@@ -607,6 +617,25 @@ namespace ttg_madness {
       }
     }
 
+    template <std::size_t i>
+    void set_global_argstream_size(std::size_t size) {
```
Comment (evaleev) [inline ttg/ttg/madness/ttg.h:621]: global -> static?

File: examples/t9/t9_streaming.cc (hunk 1 above)
Comment (pnookala) [reply examples/t9/t9_streaming.cc:378]: The comment is not valid anymore. I will clean it up.

File: ttg/ttg/madness/ttg.h (hunk 2 above)
Comment (pnookala) [reply ttg/ttg/madness/ttg.h:621]: Sure, will change it.

**Comment by evaleev** [comment]
@pnookala seems that this change breaks the existing use of streaming terminal? https://github.com/TESSEorg/ttg/runs/3020441176?check_suite_focus=true#step:10:216

**Pull request merged**